code/src/main/java/com/onlinecontacttracing/messaging/EmailSender.java

```diff
@@ -0,0 +1,116 @@
+import com.google.api.services.gmail.model.Message;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.io.ByteArrayOutputStream;
+import java.util.Properties;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
+/**
+* Email sender for sending emails to user's contacts.
+*/
+public class EmailSender {
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final List<String> SCOPES = Collections.singletonList(GmailScopes.GMAIL_SEND);
+  private static final String serviceAccountId = "online-contact-tracing@appspot.gserviceaccount.com";
+  private static final String emailToSendWith = "cccoders@onlinecontacttracing.com";
+  private static final String CREDENTIALS_FILE_PATH = "online-contact-tracing-f798898872f4.p12";
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private static final String APPLICATION_NAME = "Online Contact Tracing";
+  static final Logger log = Logger.getLogger(EmailSender.class.getName());
+
+  /**
+  * Sends emails with a email subject and a message to a list of contacts.
+  */
+  public static void sendEmailsOut(String emailSubject, CompiledMessage compiledMessage, String messageLanguage) {
+
+    try{
+      NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+
+      InputStream in = (new EmailSender()).getClass().getClassLoader().getResourceAsStream(CREDENTIALS_FILE_PATH);
+
+      GoogleCredential serviceAccountCredential = new GoogleCredential.Builder()
+        .setTransport(HTTP_TRANSPORT)
+        .setJsonFactory(JSON_FACTORY)
+        .setServiceAccountId(serviceAccountId)
+        .setServiceAccountUser(emailToSendWith)
+        .setServiceAccountScopes(Collections.singleton(GmailScopes.GMAIL_SEND))
+        .setServiceAccountPrivateKeyFromP12File(in)
+        .build();
+
+      Gmail service = new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, serviceAccountCredential)
+                  .setApplicationName(APPLICATION_NAME)
+                  .build();
+                  
+      compiledMessage.compileMessages(messageLanguage);
+      String emailBody = compiledMessage.getCompiledBackendMessage();
+
+      NotificationBatch notificationInfo = ofy().load().type(NotificationBatch.class).id(compiledMessage.getUserId()).now();
+
+      for (PersonEmail contact : notificationInfo.getPersonEmails()) {
+        sendMessage(contact.getEmail(), emailSubject, emailBody, service);
+        contact.markContactedSuccessfully();
+      }
+                  
+    } catch (MessagingException e) {
+        log.warning("error in sending emails out");
+    } catch (GeneralSecurityException e) {
+      log.warning("http transport failed, security error");
+    } catch (Exception e) {
+      log.warning("general exception");
+      e.printStackTrace();
```

Comment:
you can actually write the error to the log, please do so for all of the exception types.